packages/babel-plugin/src/utils/ast-builders.tsx

```diff
@@ -380,10 +408,14 @@ const styledTemplate = (opts: StyledTemplateOpts, meta: Metadata): t.Node => {
 export const compiledTemplate = (node: t.Expression, sheets: string[], meta: Metadata): t.Node => {
   const nonceAttribute = meta.state.opts.nonce ? `nonce={${meta.state.opts.nonce}}` : '';
 
+  const keyAttributeExpression = getKeyAttributeExpression(node);
+
   return template(
     `
-  <CC>
-    <CS ${nonceAttribute}>{%%cssNode%%}</CS>
+  <CC ${keyAttributeExpression ? buildKeyAttribute(keyAttributeExpression, '--cc') : ''}>
+    <CS ${
```

Comment:
The key should only be needed on the first node in the list (i.e. `CC`) — you should be able to remove this

Reply:
Same concern as https://github.com/atlassian-labs/compiled/pull/823#discussion_r707842373